<a href="https://colab.research.google.com/github/Saujanyapokhrel/Data-Processing-Visualization/blob/main/Quiz12.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from google.colab import auth
auth.authenticate_user()
print('Authenticated')

Authenticated


getting the user authenticated 

In [3]:
%%bigquery --project data-visualization-338722
SELECT *
FROM `bigquery-public-data.london_fire_brigade.fire_brigade_service_calls`


incident_number  ... num_pumps_attending
0      000320-01012017  ...                 2.0
1      003727-09012017  ...                 2.0
2      003762-09012017  ...                 2.0
3      006403-16012017  ...                 2.0
4      010947-26012017  ...                 1.0
...                ...  ...                 ...
32242  000284-01012017  ...                 1.0
32243  052881-28042017  ...                 1.0
32244  008677-21012017  ...                 1.0
32245  004567-11012017  ...                 1.0
32246  052773-27042017  ...                 2.0

[32247 rows x 32 columns]

navigating through bigquery-public-data.london_fire_brigade.fire_brigade_service_calls and finding that there are 32248 entries in the dataset.

In [8]:
%%bigquery --project data-visualization-338722
SELECT 
  COUNT(*) as total_rows
FROM `bigquery-public-data.london_fire_brigade.fire_brigade_service_calls`

total_rows
0       32247

total number of entries in bigquery-public-data.london_fire_brigade.fire_brigade_service_calls

In [10]:
%%bigquery --project data-visualization-338722
SELECT COUNT(*) as number_of_late_night_false_alarms
FROM `bigquery-public-data.london_fire_brigade.fire_brigade_service_calls`
WHERE incident_group = "False Alarm" AND (hour_of_call >= 22 OR hour_of_call <= 5)

number_of_late_night_false_alarms
0                               3098

there are 3098 false alarms between 10pm  to 5 am.

In [11]:
%%bigquery --project data-visualization-338722
SELECT incident_group, AVG(num_pumps_attending) as avg_pumps_attending
FROM `bigquery-public-data.london_fire_brigade.fire_brigade_service_calls`
GROUP BY incident_group

incident_group  avg_pumps_attending
0      False Alarm             1.725776
1             Fire             1.633012
2  Special Service             1.177192

getting the average numbernum_pumps_attending at the time of incident.